### ISSUE TITLE
Annotations on type parameters in stub files are sometimes ignored

### ISSUE THREAD

**Issue by atomicknight** [typetools/checker-framework#590]
Environment: Checker Framework 1.9.11, Oracle JDK 1.8.0_65, Windows 7

When working with generic methods, type annotations on type parameters in stub files seem to be ignored under certain circumstances.

Here is a test case: https://github.com/atomicknight/cf-testcases/tree/master/methodgenerics1. The test case executes two compilations, one without stub files and one with a stub file for `java.util.concurrent.ExecutorService`. The first compilation returns these errors:
```
d:\workspace\cf-testcases\methodgenerics1\src\main\java\org\example\ExecutorServiceDecorator.java:[53,43] error: [override.return.invalid] <T/*DECL*/  extends @Initialized @Nullable Object> @Initialized @NonNull Future<T extends @Initialized @Nullable Object> submit(@Initialized @NonNull ExecutorServiceDecorator this, @Initialized @NonNull Callable<T extends @Initialized @Nullable Object> p0) in org.example.ExecutorServiceDecorator cannot override <T/*DECL*/  extends @Initialized @NonNull Object> @Initialized @NonNull Future<T extends @Initialized @NonNull Object> submit(@Initialized @NonNull ExecutorService this, @Initialized @NonNull Callable<T extends @Initialized @NonNull Object> p0) in java.util.concurrent.ExecutorService; attempting to use an incompatible return type
  found   : @Initialized @NonNull Future<T extends @Initialized @Nullable Object>
  required: @Initialized @NonNull Future<T extends @Initialized @NonNull Object>

d:\workspace\cf-testcases\methodgenerics1\src\main\java\org\example\ExecutorServiceDecorator.java:[53,66] error: [override.param.invalid] <T/*DECL*/  extends @Initialized @Nullable Object> @Initialized @NonNull Future<T extends @Initialized @Nullable Object> submit(@Initialized @NonNull ExecutorServiceDecorator this, @Initialized @NonNull Callable<T extends @Initialized @Nullable Object> p0) in org.example.ExecutorServiceDecorator cannot override <T/*DECL*/  extends @Initialized @NonNull Object> @Initialized @NonNull Future<T extends @Initialized @NonNull Object> submit(@Initialized @NonNull ExecutorService this, @Initialized @NonNull Callable<T extends @Initialized @NonNull Object> p0) in java.util.concurrent.ExecutorService; attempting to use an incompatible parameter type
  found   : @Initialized @NonNull Callable<T extends @Initialized @Nullable Object>
  required: @Initialized @NonNull Callable<T extends @Initialized @NonNull Object>
```
The second compilation returns these errors:
```
d:\workspace\cf-testcases\methodgenerics1\src\main\java\org\example\ExecutorServiceDecorator.java:[53,43] error: [override.return.invalid] <T/*DECL*/  extends @Initialized @Nullable Object> @Initialized @NonNull Future<T extends @Initialized @Nullable Object> submit(@Initialized @NonNull ExecutorServiceDecorator this, @Initialized @NonNull Callable<T extends @Initialized @Nullable Object> p0) in org.example.ExecutorServiceDecorator cannot override <T/*DECL*/  extends @Initialized @Nullable Object> @Initialized @NonNull Future<T extends @Initialized @NonNull Object> submit(@Initialized @NonNull ExecutorService this, @Initialized @NonNull Callable<T extends @Initialized @NonNull Object> p0) in java.util.concurrent.ExecutorService; attempting to use an incompatible return type
  found   : @Initialized @NonNull Future<T extends @Initialized @Nullable Object>
  required: @Initialized @NonNull Future<T extends @Initialized @NonNull Object>

d:\workspace\cf-testcases\methodgenerics1\src\main\java\org\example\ExecutorServiceDecorator.java:[53,66] error: [override.param.invalid] <T/*DECL*/  extends @Initialized @Nullable Object> @Initialized @NonNull Future<T extends @Initialized @Nullable Object> submit(@Initialized @NonNull ExecutorServiceDecorator this, @Initialized @NonNull Callable<T extends @Initialized @Nullable Object> p0) in org.example.ExecutorServiceDecorator cannot override <T/*DECL*/  extends @Initialized @Nullable Object> @Initialized @NonNull Future<T extends @Initialized @NonNull Object> submit(@Initialized @NonNull ExecutorService this, @Initialized @NonNull Callable<T extends @Initialized @NonNull Object> p0) in java.util.concurrent.ExecutorService; attempting to use an incompatible parameter type
  found   : @Initialized @NonNull Callable<T extends @Initialized @Nullable Object>
  required: @Initialized @NonNull Callable<T extends @Initialized @NonNull Object>
```
Note that with the stub file, the type declaration is correctly read from the stub file, but the annotation is seemingly not applied to the return type and to the parameter types.

Discussion on the mailing list: https://groups.google.com/forum/#!topic/checker-framework-discuss/QK8USJrqkjA

**Comment by mernst** [comment]
We believe that the stub parser is correctly parsing the stub file, but the code that checks whether a method override is legal is receiving an annotated type that for some reason doesn't have the type argument.

**Comment by smillst** [comment]
Closed via aa5842f
